core/transport/src/main/proto/grpc/block_worker.proto

```diff
@@ -16,6 +16,13 @@ service BlockWorker {
    * Returns the status of the file or directory.
    */
   rpc GetStatus (grpc.file.GetStatusPRequest) returns (grpc.file.GetStatusPResponse);
+  /**
+   * If the path points to a file, the method returns a singleton with its file information.
+   * If the path points to a directory, the method returns a list with file information for the
+   * directory contents.
+   */
+  rpc ListStatus(grpc.file.ListStatusPRequest) returns (stream grpc.file.ListStatusPResponse);
```

Comment:
any reason we need a stream of responses that already contain a list?

Reply:
Please refer to the file: core/transport/src/main/proto/grpc/file_system_master.proto

My understanding is:
a stream is used to transfer large chunks of data, e.g. for read/write, the data of a read response or a write request may exceed the limit of a single RPG normal message.

The same for a ListStatus() response. The response of the ListStatus() of a large dir with a lot of entries will too large to be fixed in a normal RPC message.
So, stream must be used to transfer the result.

Reply:
I see, that make sense